src/context/EarnManagerProvider/providers/yearn/components/YearnManager/Deposit/DepositReducer.ts

```diff
@@ -0,0 +1,115 @@
+import { DepositValues } from 'context/EarnManagerProvider/components/Deposit/Deposit'
+
+import { YearnVault } from '../../../api/api'
+
+type EstimatedGas = {
+  estimatedGasCrypto?: string
+}
+
+type YearnDepositValues = DepositValues &
+  EstimatedGas & {
+    txStatus: string
+    usedGasFee: string
+  }
+
+type YearnDepositState = {
+  vault: YearnVault
+  userAddress: string | null
+  approve: EstimatedGas
+  deposit: YearnDepositValues
+  loading: boolean
+  pricePerShare: string
+  txid: string | null
+}
+
+export const initialState: YearnDepositState = {
+  txid: null,
+  vault: { apy: { net_apy: 0 } } as YearnVault,
```

Comment:
please don't cast here and create a valid empty state

Reply:
Done.